data/db-schema/src/main/resources/net/corda/db/schema/crypto/migration/crypto-creation-v1.0.xml

```diff
@@ -156,5 +179,27 @@
             <column name="tenant_id"/>
             <column name="alias"/>
         </createIndex>
+        <createTable tableName="crypto_signing_key_material">
+            <column name="wrapping_key_alias" type="VARCHAR(64)">
+                <constraints nullable="false"/>
+            </column>
+            <column name="wrapping_key_generation" type="VARCHAR(64)">
+                <constraints nullable="false"/>
+            </column>
+            <column name="signing_key_tenant_id" type="VARCHAR(12)">
+                <constraints nullable="false"/>
+            </column>
+            <column name="signing_key_id" type="CHAR(12)">
+                <constraints nullable="false"/>
+            </column>
+            <column name="signing_key_material" type="VARBINARY(1048576)">
+                <constraints nullable="false"/>
+            </column>
+        </createTable>
+        <addPrimaryKey constraintName="wrapping_key_alias, wrapping_key_generation, signing_key_tenant_id, signing_key_id"
+                       tableName="crypto_signing_key_material"
+                       columnNames="tenant_id, key_id"/>
```

Comment:
This is a bit odd.
Why not make `signing_key_id` the PK and use a UUID type for that?

The mapping etc will be much simpler and even things like retrieving a key in queries.

Reply:
Let's go to synthetic keys.